packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -162,7 +156,9 @@ export abstract class BaseTransaction {
 		this._signSignature = rawTransaction.signSignature;
 		this.timestamp = rawTransaction.timestamp;
 		this.type = rawTransaction.type;
-		this.receivedAt = rawTransaction.receivedAt || new Date();
```

Comment:
`receivedAt` is a property which is only maintained in memory and is allowed to be undefined. Transactions saved in the blockchain don't maintain this property and every node maintains it's own value for `receivedAt` property. Therefore, I think it should only be explicitly set, if the `receivedAt` property isn't provided to the constructor, it should be undefined/null.

Reply:
If we allow `receivedAt` as undefined, how should we calculate `timeElapsed` in the `isExpired` function?
it needs current time and the created time

Reply:
I guess receivedAt is only used in the transaction pool. Therefore, it's the responsibility of the transaction pool to assign receivedAt property for each transaction as soon as it enters transaction pool. It's done here: https://github.com/LiskHQ/lisk-elements/blob/development/packages/lisk-transaction-pool/src/transaction_pool.ts#L439

Reply:
so if `isExpired` is called without the receivedAt, it should throw an error?